spec/requests/vendor_api/post_make_an_offer_spec.rb

```diff
@@ -69,6 +71,44 @@
         ],
       )
     end
+
+    it 'can change a rejection into an offer' do
+      application_choice = create_application_choice_for_currently_authenticated_provider(
+        status: 'rejected',
+      )
+
+      request_body = { "data": { "conditions": ['DBS Check'] } }
+      post_api_request "/api/v1/applications/#{application_choice.id}/offer", params: request_body
+
+      expect(parsed_response).to be_valid_against_openapi_schema('SingleApplicationResponse')
+      expect(parsed_response['data']['attributes']['status']).to eq('offer')
+    end
+
+    it 'cannot update an offer once it has been accepted by the candidate' do
```

Comment:
I wonder if we need to test the invalid state transitions here, perhaps they are already tested elsewhere in the system? We should be able to rely on the state machine to avoid duplicated tests.

Reply:
Althought they share the same model, candidate_interface, provider_interface and vendor_api are different interfaces/products. These tests are there not to test if the transitions are allowed, but to test these actions can be completed through the API.

Reply:
My point is about these 4:

- `cannot update an offer once it has been accepted by the candidate`
- `cannot update an offer once it has been declined by the candidate`
- `cannot reject application if already rejected`
- `cannot reject an application that a candidate has already accepted`

We already have a test that checks the behaviour if the user attempts an invalid state change:

https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/blob/377f338bb81fdc71cf23cc72f3625965f9f70233/spec/requests/vendor_api/post_make_an_offer_spec.rb#L93-L103

It seems like these 4 tests duplicate that? We don't need specific tests for each invalid transition, since that would mean we would have to add a dozen more tests and it'll just duplicate the state machine.

Reply:
Ok